Sources/GenerateValFileTests/GenerateValFileTests.swift

```diff
@@ -0,0 +1,153 @@
+import ArgumentParser
+import Foundation
+
+/// A command-line tool that, given a list of .val files, generates XCTest cases for annotated .val
+/// files as part of our build process.
+@main
+struct GenerateValFileTests: ParsableCommand {
+
+  @Option(
+    name: [.customShort("o")],
+    help: ArgumentHelp("Write output to <file>.", valueName: "output-swift-file"),
+    transform: URL.init(fileURLWithPath:))
+  var outputURL: URL
+
+  @Option(
+    name: [.customShort("n")],
+    help: ArgumentHelp("Name of generated test case.", valueName: "XCTestCase name"))
+  var testCaseName: String
+
+  @Argument(
+    help: "Paths of annotated val source files to be tested.",
+    transform: URL.init(fileURLWithPath:)
+  )
+  var valSourceFiles: [URL]
+
+  /// Returns the Swift source of the test function for the Val at `source`, having `firstLine` as
+  /// its first line.
+  func swiftFunctionTesting(valAt source: URL, withFirstLine firstLine: Substring) throws -> String
+  {
+    let parsed = try firstLine.parsedAsFirstLineOfAnnotatedValFileTest()
+    let testID = source.deletingPathExtension().lastPathComponent.asSwiftIdentifier
+
+    return """
+
+        func test_\(parsed.methodName)_\(testID)() throws {
+          try \(parsed.methodName)(
+            \(String(reflecting: source.path)), expectSuccess: \(parsed.expectSuccess))
+        }
+
+      """
+  }
+
+  func run() throws {
+    var output =
+      """
+      import XCTest
+
+      final class \(testCaseName.asSwiftIdentifier): XCTestCase {
+
+      """
+
+    for f in valSourceFiles {
+      let firstLine = try String(contentsOf: f).split(separator: "\n", maxSplits: 1).first ?? ""
+      do {
+        output += try swiftFunctionTesting(valAt: f, withFirstLine: firstLine)
```

Comment:
```suggestion
        output += try swiftFunctionTesting(valAt: f)
```

`firstLine` is computed here only to be passed to `swiftFunctionTesting`, but the first argument is enough to compute this value in the callee. That would make its API simpler, as the call site would not have to ensure it is satisfying the relationship between `source` and `firstLine`.

Reply:
Good point; this was left over from when I was collecting all the first lines async before generating the code.